src/main/java/validation/IndentationValidity.java

```diff
@@ -1,28 +1,183 @@
 package validation;
 
+import java.util.Stack;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 /**
  * Check if the indentation is valid or not.
  */
+@SuppressWarnings("PMD.GuardLogStatement")
 public final class IndentationValidity implements Validation {
-
+	
 	public IndentationValidity(final String code) {
-		this.code = code;
+		part = code.split("\n");   	// code split
+		codeLine = part.length;			
+		validity = true;
 	}
 	
 	@Override
 	public boolean valid() {
-		throw new UnsupportedOperationException("Not implemented yet.");
+		
+		calIndentNum();			// Calculate Base Indent
+		final Stack<Integer> stack = new Stack<Integer>();
+		
+		if(isNonSpaceFirstLine()) {				// first line doesn't have indent
+			for (int i=0;i<codeLine;i++) {	
+				
+				final int num = getIndent(part[i]);	
+				
+				if(num == -1) {
+					setError("Indentation Type Miss Match", i+1);
+					validity = false;
+					return false;
+				}
+				
+				if(compareIndentNum(stack, num,i)) {continue;}
+				else {return false;}
+				
+			}
+		} else {
+			
+			setError("First Line Do Not Has Indentation", 1);
+			validity = false;
+			
+		}
+	
+		return validity;
 	}
-
 	@Override
 	public int line() {
-		throw new UnsupportedOperationException("Not implemented yet.");
+		if(validity) {
+			throw new IllegalStateException();
+		}else {
+			return errorLine;			
+		}
 	}
-
 	@Override
 	public String reason() {
-		throw new UnsupportedOperationException("Not implemented yet.");
+		if(validity) {
+			throw new IllegalStateException();
+		}
+		else {
+			return error;
+		}
+	}
+	private boolean isNonSpaceFirstLine() {
+		final Pattern patternFirstSpace = Pattern.compile(nonSpacePattern);	
+		final Matcher matcher = patternFirstSpace.matcher(part[0]);
+		return matcher.find();
+	}
+	private int findIndentLine() {
```

Comment:
The method name should be noun because the return type is not void.
How about 'firstIndentLine'?